### PR TITLE
Migration info table

### PR DESCRIPTION
**Pull request by tisnik**
# Description

Migration info table

## Type of change

- New feature (non-breaking change which adds functionality)
- Unit tests (no changes in the code)
- Documentation update

## Testing steps

Done on CI side.

## Checklist
* [x] `make before_commit` passes
* [x] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: storage.go [hunk 1]
```diff
@@ -555,21 +582,99 @@ func (storage DBStorage) DatabaseDropIndexes() error {
...
+}
+
+// DatabaseInitMigration method initializes migration_info table
+func (storage DBStorage) DatabaseInitMigration() error {
+	// Begin a new transaction.
+	tx, err := storage.connection.Begin()
+	if err != nil {
+		return err
+	}
+	// TODO: if the table already exists and contains the right version, return nil
```
Comment (epapbak) [inline storage.go:617]: comment can be removed

File: storage.go [hunk 2]
```diff
@@ -555,21 +582,99 @@ func (storage DBStorage) DatabaseDropIndexes() error {
...
+
+	finishTransaction(tx, err)
+
+	return err
+}
+
 // DatabaseInitialization method performs database initialization - creates all
 // tables in database.
 func (storage DBStorage) DatabaseInitialization() error {
+	// TODO: if the migration_info table already exists and contains the right version, return nil
```
Comment (epapbak) [inline storage.go:652]: can also be removed

File: mock_storage_test.go [hunk 3]
```diff
@@ -25,33 +25,35 @@ import (
...
-		databaseDropTablesCalled:        0,
-		databaseDropIndexesCalled:       0,
-		getLatestKafkaOffsetCalled:      0,
-		printNewReportsForCleanupCalled: 0,
-		cleanupNewReportsCalled:         0,
-		printOldReportsForCleanupCalled: 0,
-		cleanupOldReportsCalled:         0,
-		closeCalled:                     0,
-		writeReportCalled:               0,
+		databaseInitializationMigrationCalled: 0,
```
Comment (epapbak) [inline mock_storage_test.go:45]: maybe change to databaseInitMigrationCalled

**Review by epapbak — APPROVED** [review]

File: storage.go (hunk 2 above)
Comment (tisnik) [reply storage.go]: fixed, thank you

File: mock_storage_test.go (hunk 3 above)
Comment (tisnik) [reply mock_storage_test.go]: fixed, thank you

**Pull request merged**